pkg/github/pullrequests_test.go

```diff
@@ -1720,199 +1352,196 @@ func Test_CreatePullRequest(t *testing.T) {
+						} `graphql:"repository(owner: $owner, name: $repo)"`
+					}{},
+					map[string]any{
+						"owner": githubv4.String("owner"),
+						"repo":  githubv4.String("repo"),
+						"prNum": githubv4.Int(42),
+					},
+					githubv4mock.DataResponse(
+						map[string]any{
+							"repository": map[string]any{
+								"pullRequest": map[string]any{
+									"id": "PR_kwDODKw3uc6WYN1T",
+								},
+							},
+						},
+					),
+				),
+				githubv4mock.NewMutationMatcher(
+					struct {
+						AddPullRequestReview struct {
+							PullRequestReview struct {
+								ID githubv4.ID
+							}
+						} `graphql:"addPullRequestReview(input: $input)"`
+					}{},
+					githubv4.AddPullRequestReviewInput{
+						PullRequestID: githubv4.ID("PR_kwDODKw3uc6WYN1T"),
+						Body:          githubv4.NewString("This is a test review"),
+						Event:         githubv4mock.Ptr(githubv4.PullRequestReviewEventComment),
+						CommitOID:     githubv4.NewGitObjectID("abcd1234"),
+					},
+					nil,
+					githubv4mock.DataResponse(map[string]any{}),
 				),
 			),
-			requestArgs: map[string]interface{}{
-				"owner":       "owner",
-				"repo":        "repo",
-				"pull_number": float64(1),
-				"body":        "Great stuff!",
-				"commit_id":   "6dcb09b5b57875f334f61aebed695e2e4193db5e",
-				"path":        "file1.txt",
-				"line":        float64(2),
-				"side":        "RIGHT",
-			},
-			expectError:     false,
-			expectedComment: mockComment,
-		},
-		{
-			name: "successful reply using in_reply_to",
-			mockedClient: mock.NewMockedHTTPClient(
-				mock.WithRequestMatchHandler(
-					mock.PostReposPullsCommentsByOwnerByRepoByPullNumber,
-					http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
-						w.WriteHeader(http.StatusCreated)
-						err := json.NewEncoder(w).Encode(mockReply)
-						if err != nil {
-							http.Error(w, err.Error(), http.StatusInternalServerError)
-							return
-						}
-					}),
-				),
-			),
-			requestArgs: map[string]interface{}{
-				"owner":       "owner",
-				"repo":        "repo",
-				"pull_number": float64(1),
-				"body":        "Good point, will fix!",
-				"in_reply_to": float64(123),
+			requestArgs: map[string]any{
+				"owner":      "owner",
+				"repo":       "repo",
+				"pullNumber": float64(42),
+				"body":       "This is a test review",
+				"event":      "COMMENT",
+				"commitID":   "abcd1234",
 			},
-			expectError:     false,
-			expectedComment: mockReply,
+			expectToolError: false,
 		},
 		{
-			name: "comment creation fails",
-			mockedClient: mock.NewMockedHTTPClient(
-				mock.WithRequestMatchHandler(
-					mock.PostReposPullsCommentsByOwnerByRepoByPullNumber,
-					http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
-						w.WriteHeader(http.StatusUnprocessableEntity)
-						w.Header().Set("Content-Type", "application/json")
-						_, _ = w.Write([]byte(`{"message": "Validation Failed"}`))
-					}),
+			name: "failure to get pull request",
```

Comment:
I didn't do these GQL error cases in all the other tests intentionally yet, I just wanted to get a feel for whether it made sense. Every handler follows the same pattern on `if err := ... ; err != nil`